crash-handler-process/platforms/process-win.cpp

```diff
@@ -64,14 +86,100 @@ bool Process_WIN::isCritical(void) {
 	return critical;
 }
 
+void Process_WIN::startMemoryDumpMonitoring(const std::wstring& eventName, const std::wstring& eventFinishedName, const std::wstring& dumpPath) {
+	if (memorydump && memorydump->joinable()) {
+		return;
+	}
+
+	mds = OpenEvent(EVENT_ALL_ACCESS, FALSE, eventName.c_str());
+	if (!mds || mds == INVALID_HANDLE_VALUE) {
+		log_info << "Failed to open event for memory dump " << GetLastError()<< std::endl;
+		return;
+	}
+
+	PSECURITY_DESCRIPTOR securityDescriptor = (PSECURITY_DESCRIPTOR) LocalAlloc(LPTR, SECURITY_DESCRIPTOR_MIN_LENGTH);
+	if (InitializeSecurityDescriptor(securityDescriptor, SECURITY_DESCRIPTOR_REVISION)) {
+		if (SetSecurityDescriptorDacl( securityDescriptor, TRUE, NULL, FALSE)) {
+			SECURITY_ATTRIBUTES eventSecurityAttr = {0};
+			eventSecurityAttr.nLength = sizeof(SECURITY_ATTRIBUTES);
+			eventSecurityAttr.lpSecurityDescriptor = securityDescriptor;
+			eventSecurityAttr.bInheritHandle = FALSE;
+
+			mdf = CreateEvent( &eventSecurityAttr, TRUE, FALSE, eventFinishedName.c_str());
+			if (!mdf || mdf == INVALID_HANDLE_VALUE) {
+				log_info << "Failed to create event for memory dump finish " << GetLastError()<< std::endl;
+			}
+		}
+	}
+	LocalFree(securityDescriptor);
+	memorydumpPath = dumpPath;
+	memorydump = new std::thread(&Process_WIN::memorydump_worker, this);
+}
+
+const std::wstring generateCrashDumpFileName()
+{
+	std::time_t t = std::time(nullptr);
+	wchar_t wstr[100];
+	std::wstring file_name = L"crash_memory_dump.dmp";
+	if (std::wcsftime(wstr, 100, L"%Y%M%d_%H%M%S.dmp", std::localtime(&t))) {
+		file_name = wstr;
+	}
+	return file_name;
+}
+
+void Process_WIN::memorydump_worker() {
+	log_info << "Memory dump worker started" << std::endl;
+	HANDLE handles[] = { hdl, mds };
+	DWORD ret = WaitForMultipleObjects(2, handles, FALSE, INFINITE);
+	if (ret - WAIT_OBJECT_0 == 1) {
+		log_info << "Memory dump worker event recieved" << std::endl;
+		if (std::filesystem::exists(memorydumpPath) && UploadWindow::getInstance()->createWindow()) {
+			alive = false;
+			UploadWindow::getInstance()->crashCaught();
+			log_info << "Window created. Waiting for user decision" << std::endl;
+			if (UploadWindow::getInstance()->waitForUserChoise() == IDYES) {
+				log_info << "User selected OK for saving a dump" << std::endl;
+				UploadWindow::getInstance()->savingStarted();
+
+				const std::wstring file_name = generateCrashDumpFileName();
+				UploadWindow::getInstance()->setDumpFileName(file_name);
+
+				bool dump_saved = Util::saveMemoryDump(PID, memorydumpPath, file_name);
+				if (dump_saved) {
+					UploadWindow::getInstance()->savingFinished();
+					if (UploadWindow::getInstance()->waitForUserChoise() == IDYES) {
+						Util::uploadToAWS(memorydumpPath, file_name);
+						UploadWindow::getInstance()->waitForUserChoise();
+						Util::removeMemoryDump(memorydumpPath, file_name);
```

Comment:
Should we check for the return value of `removeMemoryDump` to log the error in case we were not able to remove it?

Reply:
most of the time it will be caused by something outside of our control. like user removed file manually or blocked by opening in some app. 
but yes such logs can be useful for situation if we have some bug. 
I will update code.

Reply:
Nice ty